examples/generative/diffusion/generate_dfsr.py

```diff
@@ -0,0 +1,942 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle  # TODO remove
+
+import hydra
+import numpy as np
+import PIL.Image
+import torch
+import torchvision.utils as tvu
+import torch.nn.functional as F
+import torchvision.transforms as transforms
+import tqdm
+from omegaconf import DictConfig
+from utils import StackedRandomGenerator, open_url
+
+from modulus.distributed import DistributedManager
+from modulus.launch.logging import PythonLogger, RankZeroLoggingWrapper
+from utils import EasyDict, construct_class_by_name
+import copy
+import logging
+import matplotlib.pyplot as plt
+from einops import rearrange
+from mpl_toolkits.axes_grid1 import ImageGrid
+import math
+import pickle
```

Comment:
Generally we like to avoid pickle library due to its security concerns. Might be fine inside an example directory though...